tidy3d/plugins/microwave/impedance_calculator.py

```diff
@@ -0,0 +1,63 @@
+"""Class for computing characteristic impedance of transmission lines."""
+
+from __future__ import annotations
+
+
+import pydantic.v1 as pd
+import numpy as np
+from typing import Optional
+from ...components.data.monitor_data import FieldData, FieldTimeData, ModeSolverData
+
+from ...components.base import Tidy3dBaseModel
+from ...components.validators import ValidationError
+
+from .path_integrals import VoltageIntegralAA, CurrentIntegralAA
+
+
+class ImpedanceCalculator(Tidy3dBaseModel):
+    """Tool for computing the characteristic impedance of a transmission line."""
+
+    voltage_integral: Optional[VoltageIntegralAA] = pd.Field(
+        ...,
+        title="Voltage Integral",
+        description="Integral for computing voltage.",
+    )
+
+    current_integral: Optional[CurrentIntegralAA] = pd.Field(
+        ...,
+        title="",
+        description="Integral for computing current.",
+    )
+
+    def compute_impedance(self, em_field: FieldData | ModeSolverData | FieldTimeData):
+        # If both voltage and current integrals have been defined then impedance is computed directly
+        if self.voltage_integral:
+            voltage = self.voltage_integral.compute_voltage(em_field.field_components)
+        if self.current_integral:
+            current = self.current_integral.compute_current(em_field.field_components)
+
+        # If only one of the integrals has been provided then fall back to using total power (flux)
+        # with Ohm's law. The input field should cover an area large enough to render the flux computation accurate.
+        if not self.voltage_integral:
+            flux = em_field.flux
+            if isinstance(em_field, FieldTimeData):
+                voltage = flux / current
+            else:
+                voltage = 2 * flux / np.conj(current)
```

Comment:
is the difference of field being FieldTimeData or not coming from whether the field is real/complex?

Reply:
That seems like the same condition to me. The difference between the computations comes from either considering the instantaneous power (time-domain/real) vs  time-averaged power (freq domain/complex).

Reply:
I see. Will be good to add a explanation here.